flowmachine/flowmachine/core/init.py

```diff
@@ -10,35 +10,201 @@
 From a developer perspective, this is where one-time operations
 should live - for example configuring loggers.
 """
+import warnings
+from contextlib import contextmanager
 
 import redis
 import structlog
-import warnings
 from concurrent.futures import ThreadPoolExecutor
-from typing import Union
+from typing import Tuple, Optional
+
+from redis import StrictRedis
 
 import flowmachine
-from flowmachine.core import Connection, Query
+from flowmachine.core import Connection
+from flowmachine.core.context import bind_context, context, get_db
+from flowmachine.core.errors import NotConnectedError
 from flowmachine.core.logging import set_log_level
 from get_secret_or_env_var import environ, getenv
 
 logger = structlog.get_logger("flowmachine.debug", submodule=__name__)
 
 
+@contextmanager
+def connections(
+    *,
+    log_level: Optional[str] = None,
+    flowdb_port: Optional[int] = None,
+    flowdb_user: Optional[str] = None,
+    flowdb_password: Optional[str] = None,
+    flowdb_host: Optional[str] = None,
+    flowdb_connection_pool_size: Optional[int] = None,
+    flowdb_connection_pool_overflow: Optional[int] = None,
+    redis_host: Optional[str] = None,
+    redis_port: Optional[int] = None,
+    redis_password: Optional[str] = None,
+    conn: Optional[Connection] = None,
+) -> None:
+    """
```

Comment:
Lot of doctoring duplication between these, which is annoying, but I think better than just reffing between them or assigning `__doc__` on balance.